server/src/main/java/org/opensearch/search/startree/StarTreeQuery.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SegmentReader;
+import org.apache.lucene.search.ConstantScoreScorer;
+import org.apache.lucene.search.ConstantScoreWeight;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryVisitor;
+import org.apache.lucene.search.ScoreMode;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.util.Accountable;
+import org.opensearch.common.lucene.Lucene;
+import org.opensearch.index.codec.composite.CompositeIndexFieldInfo;
+import org.opensearch.index.codec.composite.CompositeIndexReader;
+import org.opensearch.index.compositeindex.datacube.startree.index.StarTreeValues;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/** Query class for querying star tree data structure */
+public class StarTreeQuery extends Query implements Accountable {
+
+    /**
+     * Star tree field info
+     * This is used to get the star tree data structure
+     */
+    CompositeIndexFieldInfo starTree;
+
+    /**
+     * Map of field name to a list of predicates to be applied on that field
+     * This is used to filter the data based on the predicates
+     */
+    Map<String, List<Predicate<Long>>> compositePredicateMap;
+
+    public StarTreeQuery(CompositeIndexFieldInfo starTree, Map<String, List<Predicate<Long>>> compositePredicateMap) {
+        this.starTree = starTree;
+        this.compositePredicateMap = compositePredicateMap;
+    }
+
+    @Override
+    public String toString(String field) {
+        return null;
```

Comment:
nit : lets fix this to contain actual info

An example :
```
@Override
    public String toString(String field) {
        return "BinaryDocValuesRangeQuery(fieldName=" + field + ",from=" + originalFrom + ",to=" + originalTo + ")";
    }
```

Reply:
This method is used to `Prints a user-readable version of this query.` - since this is not exposed to user and is utilized internally, I'm planning to return null here, and for OriginalOrStarTreeQuery class - I am thinking of printing the originalQuery::toString. How does this sounds?

Reply:
Resolving as not valid anymore.